### ISSUE TITLE
Requests to HTTP server freeze when using GraalVM 22.2 native images

### ISSUE THREAD

**Issue by dradosevic** [micronaut-projects/micronaut-core#8237]
### Expected Behavior

Service endpoints in Micronaut OCI showcase app (https://github.com/oracle-quickstart/oci-micronaut) return valid responses to http requests.

### Actual Behaviour

Requests to api service in Micronaut OCI showcase app freeze. These are the logs in api service on these requests:

```
13:46:39.152 [default-nioEventLoopGroup-1-3] DEBUG i.n.channel.DefaultChannelPipeline - Discarded inbound message io.micronaut.http.netty.stream.EmptyHttpRequest(DefaultHttpRequest(decodeResult: success, version: HTTP/1.1)
GET /api/catalogue HTTP/1.1
Host: localhost:8080
User-Agent: curl/7.79.1
Accept: */*) that reached at the tail of the pipeline. Please check your pipeline configuration.
13:46:39.152 [default-nioEventLoopGroup-1-3] DEBUG i.n.channel.DefaultChannelPipeline - Discarded message pipeline : [idle-state, http-server-codec, flow-control-handler, http-keep-alive-handler, http-compressor, http-decompressor, WebSocketServerCompressionHandler, http-streams-codec, chunk-writer, micronaut-http-decoder, micronaut-http-encoder, websocket-upgrade-handler, micronaut-inbound-handler, DefaultChannelPipeline$TailContext#0]. Channel : [id: 0x79b470df, L:/127.0.0.1:8080 - R:/127.0.0.1:52728].
```

### Steps To Reproduce

1. Checkout the Micronaut OCI showcase app from this commit (this commit contains upgrades to Micronaut 3.7.2 and Micronaut Gradle Plugin 3.6.2)- https://github.com/oracle-quickstart/oci-micronaut/commit/a41e102f01277fc0eb28c1052e03f37d6e760a07
2. add to /src/catalogue/src/main/resources/application.yml:
 ```yaml
datasources:
  default:
    dialect: oracle
    url: jdbc:oracle:thin:@localhost:1521/xe
    driverClassName: oracle.jdbc.OracleDriver
    username: system
    password: oracle
```
3. `cd src/catalogue`
4. start db `docker run -d -p 1521:1521 -e ORACLE_PASSWORD=oracle gvenzl/oracle-xe`
5. start catalogue service`./gradlew run --args="-Dmicronaut.server.port=8082"`
6. add to /src/api/src/main/resources/application.yml:
```yaml
http:
    client:
      follow-redirects: false
    services:
      mushop-catalogue:
        url: http://localhost:8082
```
7. `cd ../api`
8. `./gradlew nativeRun`
9.  try `curl -i http://localhost:8080/api/catalogue` - the request hangs

Notes:
- when `implementation("io.netty:netty-all:4.1.83.Final")` dependency is added to /src/api/build.gradle , the requests work without problems, which indicates that there is an issue with netty 4.1.82 used in micronaut 3.7.2
- the issue occurs only for native image, and it started to occur on GraalVM 22.2 (on GraalVM 22.1 it worked ok) 
- the issue occurs for api service (which is a gateway service for other services) only
- related to #8229

### Environment Information

JDK: 11
GraalVM: 22.2

### Example Application

https://github.com/oracle-quickstart/oci-micronaut/commit/a41e102f01277fc0eb28c1052e03f37d6e760a07

### Version

3.7.2

**Comment by graemerocher** [comment]
This is fixed by upgrading to the latest version of Netty

**Comment by yawkat** [comment]
Does that mean I don't need to look at it? looks like a pain to reproduce :D

**Comment by graemerocher** [comment]
It was fixed in 3.7.3
